app/furniture/marketplace/square_order.rb

```diff
@@ -0,0 +1,167 @@
+    end
+
+    # NOTE: Square requires that orders must include fulfillments to display
+    # in the Seller Dashboard
+    # See: https://developer.squareup.com/docs/orders-api/create-orders
+    private def prepare_square_create_order_body
+      location_id = @marketplace.settings["square_location_id"]
+      customer_id = @shopper.id
+
+      line_items = @ordered_products.map { |ordered_product|
+        {
+          name: ordered_product.name,
+          quantity: ordered_product.quantity.to_s,
+          item_type: "ITEM", # ITEM|CUSTOM_AMOUNT|CGI
+          base_price_money: {
+            amount: ordered_product.product.price.cents,
+            currency: "USD"
+          }
+        }
+      }
+
+      taxes = @marketplace.tax_rates.map { |tax_rate|
+        {
+          uid: tax_rate.id,
+          name: tax_rate.label,
+          percentage: tax_rate.tax_rate.to_s,
+          scope: "LINE_ITEM"
+        }
+      }
+
+      {
+        order: {
+          location_id:,
+          line_items:,
+          taxes:,
+          fulfillments: [
+            {
+              type: "DELIVERY", # DELIVERY|SHIPMENT|PICKUP
+              state: "PROPOSED", # PROPOSED|RESERVED|PREPARED|COMPLETED|CANCELED|FAILED
+              delivery_details: {
+                recipient: {
+                  display_name: @shopper.person.display_name,
+                  phone_number: @order.contact_phone_number,
+                  address: {
+                    address_line_1: @order.delivery_address
+                  }
+                },
+                schedule_type: "SCHEDULED", # SCHEDULED|ASAP
+                # Square requires this field. Until we have a delivery
+                # calculation, let's put `now`?
+                deliver_at: Time.now.to_datetime.rfc3339
+              }
+            }
+          ],
+          customer_id:
+        },
+        idempotency_key: square_idemp_key
+      }
+    end
+
+    # NOTE: For payments that are received outside of Square - such as Convene
+    # orders paid with Stripe --  Square recommends putting "EXTERNAL" for the
+    # `source_id` and "OTHER" for the `external_details.type`.
+    # See: https://developer.squareup.com/docs/payments-api/take-payments/external-payments
+    #
+    # NOTE: `amount_money` much match the price total of line items in the
+    # square order (`line_items.sum(&base_price_money[:amount])`) to be valid
+    #
+    # TODO: Consider adding a price check?
+    private def prepare_square_create_order_payment_body
+      {
+        source_id: "EXTERNAL",
+        idempotency_key: square_idemp_key,
+        amount_money: {
+          amount: @order.product_total.cents,
+          currency: "USD"
+        },
+        order_id: @square_order_id,
+        location_id: @square_location_id,
+        external_details: {
+          type: "OTHER",
+          source: "CONVENE_SYSTEM_PAYMENT for Space #{@space.id}"
```

Comment:
I would say something like `Paid by Stripe (Charge #{order.stripe_payment_id}) via #{space.name} (#{space.id})` so we have more context than just the space id and we can use that to find the Stripe charge later.